elastica/boundary_conditions.py

```diff
@@ -1,70 +1,135 @@
 __doc__ = """ Numba implementation module for boundary condition implementations that constrain or
 define displacement conditions on the rod"""
-__all__ = ["FreeRod", "OneEndFixedRod", "HelicalBucklingBC"]
+__all__ = [
+    "ConstraintBase",
+    "FreeBC",
+    "FreeRod",  # Deprecated: remove v0.3.0
+    "OneEndFixedBC",
+    "OneEndFixedRod",  # Deprecated: remove v0.3.0
+    "FixedConstraint",
+    "HelicalBucklingBC",
+]
+
+import warnings
+from typing import Optional, Type
 
 import numpy as np
-from elastica._rotations import _get_rotation_matrix
+
+from abc import ABC, abstractmethod
 
 import numba
 from numba import njit
 
+from elastica._rotations import _get_rotation_matrix
+from elastica.rod import RodBase
+
 
-class FreeRod:
-    """
-    This is the base class for displacement boundary conditions. It applies no constraints or displacements to the rod.
+class ConstraintBase(ABC):
+    """Base class for constraint and boundary condition implementation.
 
     Note
     ----
-    Every new displacement boundary condition class must be
-    derived from FreeRod class.
-    """
+    Constraint class must inherit BaseConstraint class.
 
-    def __init__(self):
-        """
-        Free rod has no input parameters.
-        """
-        pass
+        Attributes
+        -------
+        system : RodBase
+        node_indices : None or numpy.ndarray
+        element_indices : None or numpy.ndarray
 
-    def constrain_values(self, rod, time):
+    """
+
+    _system: Type[RodBase]
```

Comment:
I think this will cause problems if we try to use BC for rigid bodies. Have you tested them @skim0119 , since rigid bodies are derived from `RigidBodyBase`

Reply:
Good point. I changed to `Union[Type[RodBase], Type[RigidBodyBase]]`. In the future, we should have a shared base class for all `elastica-objects`. [e53c376](https://github.com/GazzolaLab/PyElastica/pull/56/commits/e53c376c7e690456de5818de1f8c51d8cee5dfa6)